core/src/main/java/io/undertow/server/protocol/ajp/AjpReadListener.java

```diff
@@ -215,6 +215,17 @@ public void handleEvent(AjpServerResponseConduit channel) {
             //we need to set the write ready handler. This allows the response conduit to wrap it
             responseConduit.setWriteReadyHandler(writeReadyHandler);
 
+            // Check if the number of cookies sent exceeded the maximum
+            try {
+                httpServerExchange.getRequestCookies();
```

Comment:
Thank you for the review.

> This is not a good approach for performance reasons, as this involves creating a cookie map that may not be used (e.g. if the request is just serving a resource and does not care about cookies).

I'll change to invoke `Cookies.parseRequestCookies()` directly instead of calling it via `httpServerExchange.getRequestCookies()`. This does not create a cookie map. Is this approach ok?

> Any sort of check for this would also have to be dealt with in the HTTP/2 code to keep it consistent.

Got it. I'll also update `Http2ReceiveListener.handleRequests()`, too. 

By the way, 

> ERROR [io.undertow] (default I/O-24) UT005085: Connection io.undertow.server.protocol.http2.Http2ServerConnection for exchange HttpServerExchange{ ...(snip)... } was not closed cleanly, forcibly closing connection

I noticed the above ERROR log happened when max-parameters is exceeded (and max-cookies when I tested). So, I'll also update this PR with moving the try-catch block of `Connectors.setExchangeRequestPath()` after calling 
 `dataChannel.getResponseChannel().setCompletionListener()`.